### PR TITLE
Adding azure browse functionality

### PR DESCRIPTION
**Pull request by Benjin**
Adds the ability for users to browse their SQL databases on Azure using the Azure account manager built into VS Code.

https://github.com/user-attachments/assets/46b7e12f-13cf-4929-aa01-732e1be95715

Remaining work:
* loading indicator
* multi-tenant/subscription improvements
  * elegantly handling many tenants
  * tenant filter dropdown
* telemetry for subscription load success/failure
* playwright tests

### CODE REVIEW

File: src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx [hunk 1]
```diff
@@ -1,9 +1,334 @@
...
+        setValue: (value: string | undefined) => void;
+        currentValue?: string;
+    };
+}) => {
+    const formStyles = useFormStyles();
+
+    return (
+        <div className={formStyles.formComponentDiv}>
+            <Field label={label} orientation="horizontal" required={required}>
+                <Dropdown
```
Comment (aasimkhan30) [inline src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx:309]: Can we switch this to Combobox for auto complete?  I am guessing the list can be quite long for most people. 

https://react.fluentui.dev/?path=/docs/components-combobox--docs

File: src/reactviews/README.md
Comment (aasimkhan30) [inline src/reactviews/README.md:1]: Thanks for the doc

File: src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx (hunk 1 above)
Comment (Benjin) [reply src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx:309]: Aha, good call.  I didn't realize this was a separate control, and was going to look into textbox filtering.

[This](https://react.fluentui.dev/?path=/docs/components-combobox--docs#filtering) seems like exactly what we want.

File: src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx [hunk 2]
```diff
@@ -1,9 +1,334 @@
...
+export const AzureBrowsePage = () => {
+    const context = useContext(ConnectionDialogContext);
+    const formStyles = useFormStyles();
+    const [isAdvancedDrawerOpen, setIsAdvancedDrawerOpen] = useState(false);
+
+    if (context === undefined) {
+        return undefined;
+    }
+
+    const [subscriptions, setSubscriptions] = useState<string[]>([]);
```
Comment (aasimkhan30) [inline src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx:31]: You can probably combine these states into a single object or you can move these into the dropdown component. That will make the code simpler. 

So Azure dropdown will have 2 states variable  `values` and `value`.

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 3]
```diff
@@ -33,6 +34,14 @@ import {
...
 } from "../reactviews/common/forms/form";
 import { ApiStatus } from "../sharedInterfaces/webview";
+import {
+    AzureSubscription,
+    VSCodeAzureSubscriptionProvider,
+} from "@microsoft/vscode-azext-azureauth";
+import { uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { ResourceManagementClient } from "@azure/arm-resources";
+import { getErrorMessage } from "../utils/utils";
+// import { createSubscriptionContext, IActionContext } from '@microsoft/vscode-azext-utils';
```
Comment (Benjin) [inline src/connectionconfig/connectionDialogWebviewController.ts:44]: Remove

File: src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx [hunk 4]
```diff
@@ -1,9 +1,334 @@
...
+    const [databases, setDatabases] = useState<string[]>([]);
+    const [selectedDatabase, setSelectedDatabase] = useState<
+        string | undefined
+    >(undefined);
+
+    useEffect(() => {
+        const subs = removeDuplicates(
+            context.state.azureDatabases.map((server) => server.subscription),
+        );
+        setSubscriptions(subs);
```
Comment (Benjin) [inline src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx:60]: these (and others) should be ordered.

**Review by aasimkhan30 — APPROVED** [review]

**Pull request merged**